src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayPersistence.cs

```diff
@@ -15,6 +18,11 @@ internal InMemoryGatewayPersistence()
         /// </summary>
         protected internal override void Setup(FeatureConfigurationContext context)
         {
+            if (!PersistenceStartup.HasSupportFor<StorageType.GatewayDeduplication>(context.Settings))
```

Comment:
This check being here confuses me. All of the other changes are in the actual feature that needs the storage, but this one is in the actual InMemory implementation, which clearly does support dedup.

If we want this to actually catch a problem for gateway deduplication, shouldn't this be in the actual Gateway feature instead?